dashboard/lib/build_dashboard_page.dart

```diff
@@ -218,17 +221,33 @@ class BuildDashboardPageState extends State<BuildDashboardPage> {
         ),
         onChanged: (String? selectedBranch) {
           branch = selectedBranch;
-          _updateNavigation(context);
+          _updateNavigation(context, _buildState);
         },
-        items: _buildState.branches.map<DropdownMenuItem<String>>((String value) {
-          return DropdownMenuItem<String>(
-            value: value,
+        items:
+            // FOR REVIEW: to ensure a valid transition no matter what invalid combinations of repo + branch user inputed,
```

Comment:
nit: These would work better as standalone comments instead of intending them for the reviewer. GitHub comments should be used for conversations